### PR TITLE
Feat: added verification challenge

### PR DESCRIPTION
**Pull request by maudnals**
## Fixes KILTProtocol/ticket#343

Description: Please see the ticket.

## How to test:

- Checkout this branch;
- `docsify serve .`;
- Go through the tutorial as a newcomer would. Check that everything works and makes sense - focus on the `Verification: do it right` step.

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

NOTE: this PR includes minor changes on code display that were needed for consistency with the new "verification-right" step. Please treat these as minor in your review (understand: ignore them 🙂 )!

### CODE REVIEW

**Comment by weichweich** [comment]
The link inside the README.md does not point to setup (`👉 [Step 0: Setup](verification)`). I think this is not related to this PR but could be solved by this PR?

File: verification-right.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+Paste the following code into it (make sure to replace `<dataToVerify>` and `<challenge>` with the relevant objects):
+
+```javascript
+const Kilt = require('@kiltprotocol/sdk-js')
+
+const { proof, attestedClaimStruct } = JSON.parse(<dataToVerifyJSON>)
+
+// verify the proof (<challenge> is the uuid you've generated as the verifier)
+const isSenderOwner = Kilt.Crypto.verify(<challenge>, proof, attestedClaimStruct.attestation.owner)
```
Comment (weichweich) [inline verification-right.md]: I think this should be `const isSenderOwner = Kilt.Crypto.verify(challenge, proof, attestedClaimStruct.request.claim.owner)`

File: verification-right.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+## As the <span class="label-role verifier">verifier</span>: verify the `proof` and `attestedClaim`
+
+Create a new file `verification-with-challenge.js`.
+
+Paste the following code into it (make sure to replace `<dataToVerify>` and `<challenge>` with the relevant objects):
+
+```javascript
+const Kilt = require('@kiltprotocol/sdk-js')
+
+const { proof, attestedClaimStruct } = JSON.parse(<dataToVerifyJSON>)
```
Comment (weichweich) [inline verification-right.md]: everywhere else we use `JSON.parse(JSON.stringify(...))` except here? Is there a reason for that?

File: verification-right.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+# with yarn
+yarn add uuid
+# or with npm
+npm install uuid
+```
+
+Create a new file `challenge.js`, and paste the following code into it:
+
+```javascript
+import { v4 as uuid } from 'uuid'
```
Comment (weichweich) [inline verification-right.md]: This should be a `require` and no `import`?

File: verification-right.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+1. The <span class="label-role verifier">verifier</span> sends a nonce to the <span class="label-role claimer">claimer</span>.
+2. The <span class="label-role claimer">claimer</span> sends back this nonce signed with their **private** key, together with their `attestedClaim`.
+3. The <span class="label-role verifier">verifier</span> checks the following:
+   * Does the signature on the nonce match the public key contained in the `attestedClaim`? If so: the entity/person who just sent the `attestedClaim` plus the signed nonce is also the owner of this `attestedClaim`. If not: the `attestedClaim` might be stolen.
+   * Is the data valid? Is the attestation on-chain and not revoked? See the simple [Verification](verification).
+
+OK, let's see this in action.
+
+## As the <span class="label-role verifier">verifier</span>: create a challenge
```
Comment (weichweich) [inline verification-right.md]: Just a minor think:

challenge -> nonce

proof -> signed nonce

Comment (weichweich) [inline verification-right.md]: Just a minor think:

challenge -> nonce

proof -> signed nonce

File: verification-right.md (hunk 3 above)
Comment (maudnals) [reply verification-right.md]: 👍 right, fixed

File: verification-right.md (hunk 4 above)
Comment (maudnals) [reply verification-right.md]: 👍 that's better, done

Comment (maudnals) [reply verification-right.md]: 👍 that's better, done

File: verification-right.md (hunk 1 above)
Comment (maudnals) [reply verification-right.md]: right!! fixed

File: verification-right.md (hunk 2 above)
Comment (maudnals) [reply verification-right.md]: Uh, these `JSON.parse(JSON.stringify...` didn't make sense - cleaned up.

**Pull request merged**